core/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -58,10 +60,12 @@
    * @param file the file the block belongs to
    * @param opType the OutStream's write type
    * @param blockIndex the index of the block in the file
+   * @param tachyonConf the TachyonConf instance for this file output stream.
    * @throws IOException
    */
-  BlockOutStream(TachyonFile file, WriteType opType, int blockIndex) throws IOException {
-    super(file, opType);
+  BlockOutStream(TachyonFile file, WriteType opType, int blockIndex, TachyonConf tachyonConf)
```

Comment:
@hsaputra From your detailed design doc, I understand that you want different clients to have different configurations, but I didn't get it when I see so many modifications to function definition or constructors just to add TachyonConf as a new parameter, if it is there just for use of its getInt, getList, etc. methods, maybe we can make these methods static? I may misunderstand your design decision, please feel free to point out.